modules/core/src/shaderlib/picking/picking.js

```diff
@@ -3,20 +3,63 @@ import {picking, createModuleInjection} from '@luma.gl/core';
 createModuleInjection('picking', {
   hook: 'vs:DECKGL_FILTER_COLOR',
   injection: `
+  if (picking_uDrawZ) {
+    picking_setZ(geometry.position.z, geometry.pickingColor);
+  } else {
     picking_setPickingColor(geometry.pickingColor);
+  }
 `
 });
 
 createModuleInjection('picking', {
   hook: 'fs:DECKGL_FILTER_COLOR',
   order: 99,
   injection: `
+  if (picking_uDrawZ) {
+    color = picking_getZColor();
+  } else {
     // use highlight color if this fragment belongs to the selected object.
     color = picking_filterHighlightColor(color);
 
     // use picking color if rendering to picking FBO.
     color = picking_filterPickingColor(color);
-  `
+  }
+`
 });
 
-export default picking;
+export default {
```

Comment:
Could this be merged back to luma.gl / shadertools as part of the generic color picking shader module?

Reply:
I believe so.

Reply:
The `pack` module is in `@luma.gl/effects/experimental`. Shall we move it into `shadertools`? (the shadow module is using a similar technique) @tsherif